crates/storage/nippy-jar/src/consistency.rs

```diff
@@ -0,0 +1,187 @@
+use crate::{writer::OFFSET_SIZE_BYTES, NippyJar, NippyJarError, NippyJarHeader};
+use std::{
+    cmp::Ordering,
+    fs::{File, OpenOptions},
+    io::{BufWriter, Seek, SeekFrom},
+    path::Path,
+};
+
+/// Performs consistency checks on the [`NippyJar`] file
```

Comment:
would be nice to clarify that it also heals the inconsistencies

```suggestion
/// Performs consistency checks on the [`NippyJar`] file
```